### PR TITLE
feat[template]: get preset icons

### PR DESCRIPTION
**Pull request by DNALOCK**

### CODE REVIEW

File: api/controller/template_controller.go [hunk 1]
```diff
@@ -155,3 +157,43 @@ func (tc *TemplateController) DeleteTemplate(c *gin.Context) {
...
+
+// GetPresetIconList @Summary Delete template
+// @Description Delete a workspace template.
+// @Tags templates
+// @Produce json
+// @Param id path uint true "Template ID"
+// @Success 200 {object} model.SuccessResponse
+// @Failure 400 {object} model.ErrorResponse "Request Parse Error"
+// @Failure 500 {object} model.ErrorResponse "Unexpected System Error"
+// @Router /templates/{id} [delete]
```
Comment (StevenDB995) [inline api/controller/template_controller.go:169]: Wrong API doc. Shouldn't be a DELETE request

File: api/controller/template_controller.go [hunk 2]
```diff
@@ -155,3 +157,43 @@ func (tc *TemplateController) DeleteTemplate(c *gin.Context) {
...
+func (tc *TemplateController) GetPresetIconList(c *gin.Context) {
+	fileList, err := listFilesInDirectory("./website/dist/icons/")
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to retrieve file list"})
```
Comment (StevenDB995) [inline api/controller/template_controller.go:173]: Use model.ErrorResponse instead of gin.H

**Review by StevenDB995 — CHANGES_REQUESTED** [review]
Some changes required and need to integrate with Quanchi's template functionalities before merging into main

**Review by StevenDB995 — APPROVED** [review]

**Review by edwardlin1695 — APPROVED** [review]
Changes look good

**Pull request merged**